File: server/common/protocol/winners_message.py
from common.protocol.message import Message


class WinnersMessage(Message):
    WINN_OP = bytes([0x06])
    

    def serialize(self) -> bytes:
        return self._build_header(self.WINN_OP, 0)
    
    def should_ack(self) -> bool:
        return False

File: client/common/protocol/winners_message.go
package protocol

import "errors"


type Winners struct {}

func (win *Winners) ShouldAck() bool {
	return false
}

func (win *Winners) Serialize() []byte {
	return buildHeader([]byte{WINN_OP}, 0)
}

func (win *Winners) Deserialize(stream []byte) error{
	if !compareStreams(win.Serialize(), stream) {
		return errors.New("invalid Winners message")
	}
	return nil
}